app/http.php

```diff
@@ -0,0 +1,1039 @@
+            case 409: // Error allowed publicly
+            case 412: // Error allowed publicly
+            case 425: // Error allowed publicly
+            case 429: // Error allowed publicly
+            case 501: // Error allowed publicly
+            case 503: // Error allowed publicly
+                $code = $error->getCode();
+                break;
+            default:
+                $code = 500; // All other errors get the generic 500 server error status code
+        }
+
+        $output = [
+            'message' => $error->getMessage(),
+            'code' => $error->getCode(),
+            'file' => $error->getFile(),
+            'line' => $error->getLine(),
+            'trace' => $error->getTrace(),
+            'version' => App::getEnv('OPEN_RUNTIMES_EXECUTOR_VERSION', 'UNKNOWN')
+        ];
+
+        $response
+            ->addHeader('Cache-Control', 'no-cache, no-store, must-revalidate')
+            ->addHeader('Expires', '0')
+            ->addHeader('Pragma', 'no-cache')
+            ->setStatusCode($code);
+
+        $response->json($output);
+    });
+
+App::init()
+    ->inject('request')
+    ->action(function (Request $request) {
+        $secretKey = \explode(' ', $request->getHeader('authorization', ''))[1] ?? '';
+        if (empty($secretKey) || $secretKey !== App::getEnv('OPEN_RUNTIMES_EXECUTOR_SECRET', '')) {
+            throw new Exception('Missing executor key', 401);
+        }
+    });
+
+/** @phpstan-ignore-next-line */
+Co\run(
+    function () use ($register) {
+        $orchestrationPool = $register->get('orchestrationPool');
+        $activeRuntimes = $register->get('activeRuntimes');
+
+        /**
+         * Remove residual runtimes
+         */
+        Console::info('Removing orphan runtimes...');
+        try {
+            $connection = $orchestrationPool->pop();
+            $orchestration = $connection->getResource();
+            $orphans = $orchestration->list(['label' => 'openruntimes-executor=' . System::getHostname()]);
+        } finally {
+            isset($connection) && $orchestrationPool->push($connection);
+        }
+
+        if (\count($orphans) === 0) {
+            Console::log("No orphan runtimes found.");
+        }
+
+        $callables = [];
+        foreach ($orphans as $runtime) {
+            $callables[] = function () use ($runtime, $orchestrationPool) {
+                try {
+                    $connection = $orchestrationPool->pop();
+                    $orchestration = $connection->getResource();
+                    $orchestration->remove($runtime->getName(), true);
+                    Console::success("Successfully removed {$runtime->getName()}");
+                } catch (\Throwable $th) {
+                    Console::error('Orphan runtime deletion failed: ' . $th->getMessage());
+                } finally {
+                    isset($connection) && $orchestrationPool->push($connection);
+                }
+            };
+        }
```

Comment:
We seem to be doing something very similar in L206
Can we reuse the code ?

Reply:
Good catch. Reusing the function now.